### DISCUSSION TITLE
Will the new DE still support gnome extensions?

### DISCUSSION THREAD

**Discussion by xNS5** [pop-os/cosmic-epoch#919] [General]
I'm pretty particular about how I want my DE to work, and I've been able to achieve that using a number of gnome extensions.

**Comment by mmstick** [comment]
No, it's not based on GNOME.

**Reply by xNS5** [reply]
Got it, thanks! I can live without the majority of the gnome shell extensions I have running, but the must-have for me is this one https://github.com/velitasali/gnome-shell-extension-awesome-tiles. It's basically the only equivalent to Magnet/Rectangle on MacOS that I've found. 

I know that the new DE is still in alpha, but offhand do you know how challenging it would be to build an application that essentially replicates the functionality of this extension?

**Reply by mmstick** [reply]
Dynamic auto-tiling is built into COSMIC, on a per-display and per-workspace basis, so I don't think that'd be necessary.

**Reply by xNS5** [reply]
Awesome tiles is more of a window snapping extension with keybindings, along the same lines as Fancy Zones on Windows or Magnet/Rectangle on MacOS. I like to think of it as a more "passive" window manager which moves windows on demand compared to something that auto tiles. I could take a stab at making an application that accomplishes what I'm looking for (I've been wanting to try making something in Rust), but I wouldn't really know where to start -- especially with COSMIC still being in alpha.
